modules/programs/hexchat.nix

```diff
@@ -0,0 +1,262 @@
+        };
+      in mkOption {
+        type = nullOr (attrsOf modChannelOption);
+        default = null;
+        description = "Configures '~/.config/hexchat/servlist.conf'";
+        example = ''
+          {
+                        freenode = {
+                          autojoin = [
+                            "#home-manager"
+                            "##linux"
+                            "#nixos"
+                          ];
+                          charset = "UTF-8 (Unicode)";
+                          commands = [
+                            "ECHO Buzz Lightyear sent you a message: 'To Infinity... and Beyond!'"
+                          ];
+                          loginMethod = sasl;
+                          nickname = "my_nickname";
+                          nickname2 = "my_secondchoice";
+                          options = {
+                            accept_invalid_ssl_certificates = false;
+                            autoconnect = true;
+                            bypass_proxy = true;
+                            connect_to_selected_server_only = true;
+                            use_global_user_information = false;
+                            use_ssl_for_all_servers = false;
+                          };
+                          password = "my_password";
+                          realName = "my_realname";
+                          servers = [
+                            "chat.freenode.net"
+                            "irc.freenode.net"
+                          ];
+                          userName = "my_username";
+                        };
+                      }'';
+      };
+    conf = mkOption {
+      default = null;
+      description = "Configures '~/.config/hexchat/hexchat.conf'";
+      example = ''
+        {
+                    irc_nick1 = "mynick";
+                    irc_username = "bob";
+                    irc_realname = "Bart Simpson";
+                    text_font = "Monospace 14";
+                  };'';
+      type = with lib.types; nullOr (attrsOf str);
+    };
+    enable = mkEnableOption "HexChat - Graphical IRC client";
+    theme = mkOption {
+      default = null;
+      description = "Theme package for HexChat.";
+      # example = ""; # To-Do: Update here with a valid theme.
+      type = with lib.types; nullOr package;
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    home.packages = [ pkgs.hexchat ];
+    xdg.configFile."hexchat" = lib.mkIf (cfg.theme != null) {
+      source = cfg.theme.out;
+      recursive = true;
+    };
+    xdg.configFile."hexchat/hexchat.conf" = lib.mkIf (cfg.conf != null) {
+      force = true;
+      text = let hexchatConf = cfg.conf;
+      in concatMapStringsSep "\n" (x: x + " = " + hexchatConf.${x})
+      (attrNames hexchatConf);
+    };
+    xdg.configFile."hexchat/servlist.conf" = lib.mkIf (cfg.channels != null)
+      (if attrNames cfg.channels == [ ] then {
+        text = "";
+      } else {
+        force = true;
```

Comment:
Why?

Reply:
It's necessary to refresh servlist.conf file when channels list is empty. Or else any old value will persist and cause odd behavior. If no channel is listed it is expected to have no channel listed.

Reply:
Ah, now that I think about this, this is not a desired behavior. If the file exists, HM should raise and error for the user to either remove the file themselves or pass the `-b` parameter to make HM do it.

See: https://github.com/nix-community/home-manager#keeping-your--safe-from-harm